test_cases/solidity/insufficient_gas_griefing/relayer_fixed/relayer_fixed.sol

```diff
@@ -0,0 +1,37 @@
+/*
+ * @source: https://consensys.github.io/smart-contract-best-practices/known_attacks/#insufficient-gas-griefing
+ * @author: ConsenSys Diligence
+ * Modified by Kaden Zipfel
+ */
+
+pragma solidity ^0.5.0;
+
+contract Relayer {
+    uint transactionId;
+
+    struct Tx {
+        bytes data;
+        bool executed;
+    }
+
+    mapping (uint => Tx) transactions;
+
+    function relay(bytes memory _data, uint _gasLimit) public {
+        // replay protection; do not call the same transaction twice
+        require(transactions[transactionId].executed == false, 'same transaction twice');
+        transactions[transactionId].data = _data;
+        transactions[transactionId].executed = true;
+        transactionId += 1;
+
+        Target target = new Target();
```

Comment:
why deploy a new target everytime?

Reply:
Good point. The problem that I'm looking at is, would I have to deploy the Target contract to the mainnet and reference the address to avoid compilation errors? Or is there some way I can circumvent the compilation error?

Reply:
oh, you can probably just delete that line, and do this:

`function relay(Target target, bytes memory _data, uint _gasLimit) public {`

Reply:
Ahh, yes of course

Reply:
Fixed in https://github.com/SmartContractSecurity/SWC-registry/pull/223